sw/device/silicon_creator/lib/crypto/rsa_3072/rsa_3072_verify.c

```diff
@@ -41,6 +44,86 @@ static const uint32_t kOtbnRsaModeVerify = 1;
 static const uint32_t kOtbnRsaModeComputeRR = 2;
 static const uint32_t kOtbnRsaModeComputeM0Inv = 3;
 
+/**
+ * Message encoding prefix constant. From RFC 8017:
+ *
+ * Section 9.2, step 5:
+ *   Concatenate PS, the DER encoding T, and other padding to form
+ *   the encoded message EM as
+ *
+ *     EM = 0x00 || 0x01 || PS || 0x00 || T.
+ *
+ * Section 9.2, step 4:
+ *   Generate an octet string PS consisting of emLen - tLen - 3
+ *   octets with hexadecimal value 0xff.  The length of PS will be
+ *   at least 8 octets.
+ *
+ * Section 9.2, step 5:
+ *   Concatenate PS, the DER encoding T, and other padding to form
+ *   the encoded message EM as
+ *
+ *     EM = 0x00 || 0x01 || PS || 0x00 || T.
+ *
+ * Section 9.2, note 1:
+ *   For the nine hash functions mentioned in Appendix B.1, the DER
+ *   encoding T of the DigestInfo value is equal to the following:
+ *   ...
+ *   SHA-256: (0x)30 31 30 0d 06 09 60 86 48 01 65 03 04 02 01 05 00
+ *                04 20 || H.
+ *   ...
+ *
+ *
+ * For the specific purpose of RSA-3072 and SHA-256, some variables are fixed.
+ * In particular:
+ *   emLen = 3072 / 8 = 384 bytes
+ *   tLen = 51 bytes (19 for DigestInfo prefix above + 32 for SHA-256 digest)
+ *
+ * Therefore, the octet strings for PS and the first part of T (excluding the
+ * hash) are constant; the encoded message EM consists of simply appending the
+ * SHA-256 digest to the below prefix.
+ */
+static const uint32_t kEncodingPrefix[88] = {
```

Comment:
This uses 328 bytes for `0xffffffff`s which can be eliminated by writing them in a separate loop in `rsa_3072_encode_sha256`.